### PR TITLE
Remove body from all GET requests

### PR DESCRIPTION
**Pull request by decentralgabe**
- this was not able to be supported with the console or in any browser api

### CODE REVIEW

File: pkg/server/router/operation.go [hunk 1]
```diff
@@ -14,6 +15,11 @@ import (
 	"github.com/tbd54566975/ssi-service/pkg/service/operation"
 )
 
+const (
+	ParentParam string = "parent"
+	DoneParam   string = "done"
```
[the remark this answers is not in the document]
Comment (decentralgabe) [reply pkg/server/router/operation.go:20]: It's awkward to encode a special filter string as a query parameter. More RESTful to map it to a native query param

[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/server/router/operation.go:20]: Why is it awkward?

It's a standard pattern. See https://cloud.google.com/apis/design/naming_convention#list_filter_field

`filter` _is_ the native query param. Mapping singular fields doesn't allow any sort of advanced filtering like additional fields or boolean expressions, which we are likely to want to support in the future.

[the remark this answers is not in the document]
Comment (decentralgabe) [reply pkg/server/router/operation.go:20]: If we were using protobufs I'd agree...
Alternatively we can use a POST or PUT with a message body

[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/server/router/operation.go:20]: The decision on the API surface should be based on what's best for devs and future evolution of the API. We shouldn't allow implementation details to leak.

[the remark this answers is not in the document]
Comment (decentralgabe) [reply pkg/server/router/operation.go:20]: Can you show me how a query string might look with your suggestion?

File: pkg/server/router/operation.go [hunk 2]
```diff
@@ -77,7 +83,7 @@ func (o OperationRouter) GetOperation(c *gin.Context) {
 	framework.Respond(c, routerModel(*op), http.StatusOK)
 }
 
-type ListOperationsRequest struct {
+type listOperationsRequest struct {
```
Comment (nitro-neal) [inline pkg/server/router/operation.go:86]: You mean to make this a private Type?

File: pkg/server/router/operation.go [hunk 3]
```diff
@@ -138,19 +144,29 @@ type ListOperationsResponse struct {
...
+		if err != nil {
+			errMsg := "failed un-escaping parent param"
+			framework.LoggingRespondErrWithMsg(c, err, errMsg, http.StatusInternalServerError)
+			return
+		}
+		request.Parent = unescaped
 	}
-
+	if doneParam != nil {
+		request.Filter = fmt.Sprintf("%s = %s", DoneIdentifier, *doneParam)
```
Comment (nitro-neal) [inline pkg/server/router/operation.go:167]: if we are not using the filter as a dynamic thing someone can mess with on the front end, we should probably remove

**Review by nitro-neal — APPROVED** [review]

File: pkg/server/router/operation.go (hunk 3 above)
Comment (decentralgabe) [reply pkg/server/router/operation.go]: reworked

File: doc/swagger.yaml [hunk 4]
```diff
@@ -3060,12 +3035,11 @@ paths:
       description: List existing submissions according to a filtering query. The `filter`
         field follows the syntax described in https://google.aip.dev/160.
       parameters:
-      - description: request body
-        in: body
-        name: request
-        required: true
-        schema:
-          $ref: '#/definitions/pkg_server_router.ListSubmissionRequest'
+      - description: 'A standard filter expression conforming to https://google.aip.dev/160.
```
[the remark this answers is not in the document]
Comment (decentralgabe) [reply doc/swagger.yaml:3038]: this is related to operations not the credential submission status cc: @andresuribe87

File: pkg/server/router/operation.go [hunk 5]
```diff
@@ -138,19 +144,36 @@ type ListOperationsResponse struct {
 //	@Tags			OperationAPI
 //	@Accept			json
 //	@Produce		json
-//	@Param			request	body		ListOperationsRequest	true	"request body"
+//	@Param			parent	query		string					false	"The name of the parent's resource. For example: `parent?=/presentation/submissions`"
```
Comment (andresuribe87) [inline pkg/server/router/operation.go:147]:
```suggestion
//	@Param			parent	query		string					false	"The name of the parent's resource. For example: `?parent=/presentation/submissions`"
```

File: pkg/server/server_presentation_test.go [hunk 6]
```diff
@@ -507,21 +495,17 @@ func TestPresentationAPI(t *testing.T) {
...
 					"id":             "did:web:andresuribe.com",
 				})), holderDID, holderSigner)
 
-			request := router.ListSubmissionRequest{
-				Filter: `status = "done"`,
-			}
-
-			value := newRequestValue(ttt, request)
-			req := httptest.NewRequest(http.MethodGet, "https://ssi-service.com/v1/presentations/submissions", value)
+			query := url.QueryEscape("status=\"done\"")
```
Comment (andresuribe87) [inline pkg/server/server_presentation_test.go:505]:
```suggestion
			query := url.QueryEscape(`status="done"`)
```

**Review by andresuribe87 — APPROVED** [review]
Looks good! Some minor nits

File: doc/swagger.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (andresuribe87) [reply doc/swagger.yaml:3038]: The filters accepted are different per endpoint. 

`v1/operations` accepts values for filter like `done=true`
`v1/presentations/submissions` accepts values for filter like `status=approved`. You are correct that `status=done` is not something we'd accept.

**Pull request merged**